test/CirclesBackingFactory.t.sol

```diff
@@ -3,65 +3,638 @@ pragma solidity ^0.8.28;
+
+    // Gnosis fork ID
+    uint256 internal gnosisFork;
+
+    // The CowSwap order uid for the test instance
     bytes public uid;
 
+    // -------------------------------------------------------------------------
+    // Setup
+    // -------------------------------------------------------------------------
+
     function setUp() public {
-        gnosis = vm.createFork(vm.envString("GNOSIS_RPC"), blockNumber);
-        vm.selectFork(gnosis);
-        factory = new CirclesBackingFactory(factoryAdmin, uint256(100));
+        // Fork from Gnosis
+        gnosisFork = vm.createFork(vm.envString("GNOSIS_RPC"), FORK_BLOCK_NUMBER);
+        vm.selectFork(gnosisFork);
+
+        // Deploy factory
+        factory = new CirclesBackingFactory(FACTORY_ADMIN, 100); // test-specific "fee" if any
+
+        // Retrieve constants from factory
         VAULT = factory.VAULT();
         USDC = factory.USDC();
         CRC_AMOUNT = factory.CRC_AMOUNT();
+
+        TEST_ACCOUNT = makeAddr("alice");
+        _setAvatarWithPersonalCRC(TEST_ACCOUNT);
     }
 
-    function test_BackingFlow() public {
-        address predictedInstance = factory.computeAddress(testAccount);
+    // -------------------------------------------------------------------------
+    // Modifiers + Internal Helpers
+    // -------------------------------------------------------------------------
 
-        // first fill test account with 100 USDC
-        deal(USDC, testAccount, usdcStartAmount);
+    /**
+     * @dev This modifier sets the BACKING_ASSET state variable based on the provided index,
+     *      chosen from an array of addresses [WBTC, WETH, GNO, sDAI].
+     */
+    modifier setBackingAsset(uint8 _backingAssetIndex) {
+        address[4] memory backingAssets = [WBTC, WETH, GNO, sDAI];
 
-        // next approve factory to spend usdc
-        vm.prank(testAccount);
-        IERC20(USDC).approve(address(factory), usdcStartAmount);
+        vm.assume(_backingAssetIndex < backingAssets.length);
+        BACKING_ASSET = backingAssets[_backingAssetIndex];
+        _;
+    }
 
-        // next transfer 48CRC to factory with WETH encoded as backing asset
-        bytes memory data = abi.encode(WETH);
-        vm.prank(testAccount);
-        HUB_V2.safeTransferFrom(testAccount, address(factory), uint256(uint160(testAccount)), CRC_AMOUNT, data);
+    /**
+     * @dev Mocks CircleV1 calls to associate `_user` with a personal CRC contract
+     *      and registers the new human in HubV2 before minting personal CRC tokens.
+     *
+     * @param _user The address to set up as a Circles Human.
+     */
+    function _setAvatarWithPersonalCRC(address _user) internal {
+        address CIRCLE_USER_V1 = makeAddr("CircleUserV12");
+        // 0x29b9a7fBb8995b2423a71cC17cf9810798F6C543 -> CirclesV1
+        // 0x28d249fe -> function userToToken(address) public returns(address)
+        vm.mockCall(
+            address(0x29b9a7fBb8995b2423a71cC17cf9810798F6C543),
```

Comment:
just for me to catch up with context: why are we calling on Hub v1 for testing the lbp? are we running these tests against a fork of mainnet?

Reply:
right, for now Im happy with any solution. Overtime we can also recover and build out a re-usable mock framework for a vanilla mock deployment on chain if it s helpful

Reply:
I agree, totally make sense to use mocks in unit tests (especially, considering we are doing it on protocol lvl) instead of forking.
I always do fork before deployment to see how it will go, but @web3skeptic probably assumed it was beginning of unit tests and followed with forking pattern, a little bit of misunderstanding, would be great next time to discuss what we want to do before getting hands dirty.